### ISSUE TITLE
Reproduction Details

### ISSUE THREAD

**Issue by Arsiuuu** [google-deepmind/video_comp#2]
Dear Authors,

I have a few questions while reproducing the finetuning process and hope you can help clarify.

1. The number of training samples for each type of negative example is not uniform. How should the hierarchical pairwise preference loss mentioned in the paper be applied? For example, one video has A > B > C, while another has A > B > C > D. This seems difficult to implement with batch training.

2. For the seg_mismatch type, according to the article's meaning, the video is cropped from "query_video/start_time" to "query_video/end_time" and then the similarity is calculated with "positive_text" and "negative_text" respectively. So,  for the other negative texts corresponding to this video,   (such as temp_reorder, action_replace),  should their similarity be calculated with the original video or the cropped video?

3. How is the ​Multi-Disrupt trainingset implemented? Are there any key parameters or strategies to note? In the training process, are the negative samples ​constructed online (dynamically) or ​pre-generated offline (fixed)?

Thanks~

**Comment by mcahny** [comment]
Hi, thanks for the questions.
A dummy embedding with loss weight mask can be used to handle the varying number.
The cropped video is used for seg_mismatch evaluation.
The multi_disrupt samples are prepared offline.

**Comment by Arsiuuu** [comment]
Thank you for your answer. I still have a question: How to handle the seg_mismatch type during training? 

![Image](https://github.com/user-attachments/assets/e9368271-3da6-4465-9793-2e721d1561b9)

For example, if the original video feature is V, I need to compute a cropped video feature V' for the seg_mismatch type. Should the similarity ranking for hierarchical loss be (V, T_positive) > (V, T_temp_reorder) > (V, T_action_replace) > (V', T_seg_mismatch) > ... ? If so, how should the positive text T' (which is different from T_positive) paired with V' be used? How to define the relative magnitude of (V, T_positive) and (V', T'), since they are both correct positive pairs, only differing in the richness of the events they contain. 

Here's an example:
```

**Temp_reorder:**
  {
    "video": "training/205/oUvPl-uaqdE.mp4",
    "caption": "Preheat the oven to 250 degrees. Then, spread some olive oil evenly on a pan. Next, spread the dough on the pan. After that, spread tomato puree mixed with some salt and olive oil on the dough. Subsequently, add some basil and cut mozzarella cheese on it. Then, pour some olive oil on the top. Finally, put the preparation in the preheated oven for 15-20 minutes.",
    "negative": "Preheat the oven to 250 degrees. Then, spread the dough on the pan. First, spread some olive oil evenly on a pan. Next, spread tomato puree mixed with some salt and olive oil on the dough. After that, add some basil and cut mozzarella cheese on it. Subsequently, pour some olive oil on the top. Finally, put the preparation in the preheated oven for 15 20 minutes."
  },
**Seg_mismatch:**
  {
    "video": "training/205/oUvPl-uaqdE.mp4",
    "caption": "Pour some olive oil on the top.\\n\" \"Then, put the preparation in the preheated oven for 15-20 minutes.",**
    "positive_time": [
      123000000,
      135000000
    ],
    "negative": "Spread tomato puree mixed with some salt and olive oil on the dough.\\n\" \"Then, add some basil and cut mozzarella cheese on it.\\n\" \"Next, pour some olive oil on the top.\\n\" \"Finally, put the preparation in the preheated oven for 15-20 minutes."
  },
```

Looking forward to your answer.
